src/Benchmarks/Startup.cs

```diff
@@ -51,23 +52,35 @@ public IServiceProvider ConfigureServices(IServiceCollection services)
             // Common DB services
             services.AddSingleton<IRandom, DefaultRandom>();
             services.AddSingleton<ApplicationDbSeeder>();
-            services.AddEntityFrameworkSqlServer()
-                .AddDbContext<ApplicationDbContext>();
-
-            if (Scenarios.Any("Raw") || Scenarios.Any("Dapper"))
-            {
-                services.AddSingleton<DbProviderFactory>((provider) => {
-                    var settings = provider.GetRequiredService<IOptions<AppSettings>>().Value;
 
-                    if (settings.Database == DatabaseServer.PostgreSql)
-                    {
-                        return NpgsqlFactory.Instance;
-                    }
-                    else
+            services.AddEntityFrameworkSqlServer()
+                .AddDbContext<ApplicationDbContext>(
+                    (sp, ob) =>
+                        {
+                            var appSettings = sp.GetService<IOptions<AppSettings>>().Value;
+
+                            if (appSettings.Database == DatabaseServer.PostgreSql)
+                            {
+                                ob.UseNpgsql(appSettings.ConnectionString);
+                            }
+                            else
+                            {
+                                ob.UseSqlServer(appSettings.ConnectionString);
+                            }
+                        },
+                poolSize: 32);
```

Comment:
How should applications determine the optimal pool size?  Might a larger size be better?